reference-implementation/lib/helpers.js

```diff
@@ -58,7 +62,7 @@ exports.InvokeOrNoop = (O, P, args) => {
     throw new TypeError(`${P} is not a function`);
   }
 
-  return method.apply(O, args);
+  return Call(method, O, args);
```

Comment:
It seems including the L61-63 in the Call method you've just introduced would give them better correspondence because the lines are performing the step `If IsCallable(F) is false, throw a TypeError exception.` of the Call operation ( https://tc39.github.io/ecma262/#sec-call ).